src/amber/cli/templates/api/controller/granite/spec/controllers/{{name}}_controller_spec.cr.ecr

```diff
@@ -0,0 +1,84 @@
+<% params_name =  "#{@name.underscore}_params" -%>
+<% hash_name   =  "#{@name.underscore}_hash" -%>
+<% create_model_method  =  "create_#{@name.underscore}" -%>
+require "./spec_helper"
+
+def <%= hash_name %>
+  <%= @fields_hash.to_s %>
+end
+
+def <%= params_name %>
+  params = [] of String
+  <%- @fields_hash.keys.map do |k| -%>
+  params << "<%= k %>=#{<%= hash_name %>["<%= k %>"]}"
+  <%- end -%>
+  params.join("&")
+end
+
+def <%= create_model_method %>
+  model = <%= class_name %>.new(<%= hash_name %>)
+  model.save
+  model
+end
+
+class <%= class_name %>ControllerTest < GarnetSpec::Controller::Test
+  getter handler : Amber::Pipe::Pipeline
+
+  def initialize
+    @handler = Amber::Pipe::Pipeline.new
+    @handler.build :api do
+      plug Amber::Pipe::Error.new
+      plug Amber::Pipe::Session.new
+    end
+    @handler.prepare_pipelines
+  end
+end
+
+describe <%= class_name %>ControllerTest do
+  subject = <%= class_name %>ControllerTest.new
+
+  it "renders <%= @name %> index json" do
+    <%= class_name %>.clear
+    model = create_<%= @name.downcase %>
+    response = subject.get "/<%= Inflector.pluralize(@name) %>"
+
+    response.status_code.should eq(200)
+    response.body.should contain("Fake")
+  end
+
+  it "renders <%= @name %> show json" do
+    <%= class_name %>.clear
+    model = create_<%= @name.downcase %>
+    location = "/<%= Inflector.pluralize(@name) %>/#{model.id}"
```

Comment:
I see this snippet everywhere. Does it make sense to put the plural in the teeplate controller above?

Reply:
@robacarp Yes, yes it does.  👍